src/Search.php

```
@@ -215,16 +226,18 @@ function ($value) {
      *
      * @param float[]|int[] $values
      *
-     * @return int Index of the first occurrence of the minimum value
+     * @return int|null Index of the first occurrence of the minimum value
      */
-    private static function baseArgMin(array $values): int
+    private static function baseArgMin(array $values): ?int
     {
         $max = \min($values);
         foreach ($values as $i => $v) {
             if ($v === $max) {
                 return $i;
             }
         }
+
+        return null;
```

Note: I think it makes more sense to throw an exception here as a precondition gone wrong rather than change the return type to nullable.

---

I think so too.